packages/analytics/amplify_analytics_pinpoint/example/integration_test/events_test.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//  http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:amplify_analytics_pinpoint/amplify_analytics_pinpoint.dart';
+import 'package:amplify_analytics_pinpoint_example/amplifyconfiguration.dart';
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('recordEvent', () {
+    const boolProperty = MapEntry('boolProperty', true);
+    // Boolean properties are recorded as strings
+    const stringifiedBoolProperty = MapEntry('boolProperty', 'true');
+
+    const doubleProperty = MapEntry('doubleProperty', pi);
+    // There is some lossiness behavior to doubles
+    const lossyDoubleProperty = MapEntry('doubleProperty', 3.141592653589793);
+
+    const intProperty = MapEntry('intProperty', 42);
```

Comment:
question/suggestion: Should one of the properties of the event be random (like a UUID) so we can ensure received event came from this invocation and not from some other client?